altair/__init__.py

```diff
@@ -654,6 +655,7 @@ def __dir__():
 from altair.jupyter import JupyterChart
 from altair.expr import expr
 from altair.utils import AltairDeprecationWarning, parse_shorthand, Optional, Undefined
+from altair import typing
```

Comment:
@mattijn just want to be super clear on the scope of this, in reference to [comment](https://github.com/pola-rs/polars/pull/17995#issuecomment-2265137793).
It may even be worth adapting my notes here into the `altair.typing` module docstring.

- `altair.typing` will contain *some* public-facing types, that may be useful for extending/integrating with `altair`
	- Doesn't need to include everything
	- Some things may be useful both int/externally (e.g. `ChartType`)
- The module itself **is** accessible at the top-level, but **not** it's members
	- ✔️ `from altair.typing import EncodeKwds`
	- ✔️ `import altair.typing as alt_t; alt_t.EncodeKwds`
	- ❌ `alt.EncodeKwds` 
	- ❌ `alt.typing.EncodeKwds` 
- This is a common practice in other packages
	- https://github.com/numpy/numpy/blob/main/numpy/typing/__init__.py
	- https://github.com/numpy/numpy/blob/main/numpy/dtypes.pyi
	- https://github.com/pola-rs/polars/blob/0c2b5d82fd2726df7d7b318e5c21581a27fffdba/py-polars/polars/type_aliases.py
	- https://github.com/pandas-dev/pandas/blob/main/pandas/_typing.py

Reply:
Sounds useful to have public-facing types within `altair.typing`!